src/v/model/tests/random_batch.cc

```diff
@@ -242,7 +242,7 @@ make_random_batches(record_batch_spec spec) {
             base_sequence += num_records;
         }
         auto b = make_random_batch(batch_spec);
-        o = b.last_offset() + model::offset(num_records);
+        o = b.last_offset() + model::offset(1);
```

Comment:
Note for reviewer: I think this was a bug. It lead to gaps in the log for batches with more than one record.